run-lcb-evals.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# This script runs the LiveCodeBench custom evaluator for each of the models
+# listed in `MODELS`.
+#
+# It assumes that post-processed data in the format expected by the LiveCodeBench
+# custom evaluator: https://github.com/LiveCodeBench/LiveCodeBench?tab=readme-ov-file#custom-evaluation)
+# is stored in the following directory structure:
+#
+# data/
+# ├─ gpt-4.1-2025-04-14/
+# │  ├─ post-processed/
+# │  │  ├─ lcb-formatted.json
+# ├─ Llama-3.3-70B-Instruct/
+# │  ├─ post-processed/
+# │  │  ├─ lcb-formatted.json
+#
+# This script produces a set of LiveCodeBench evaluation files for each
+# LiveCodeBench input file in the same directory where it is located.
+#
+# Ensure you are running this script in a directory that also contains the `data`
+# folder at the same depth.
+#
+# Example usage:
+#   ./run-lcb-evals.py
+
+from dataclasses import dataclass
+
+import glob
+import subprocess
+import concurrent.futures
+
+
+@dataclass
+class LcbModelEvaluationInfo:
+    """Represents an LLM and its LiveCodeBench input files.
+
+    Attributes:
+        model (str): An LLM.
+        lcb_input_files (list[str]): A list of `.json` files conforming to the
+            format expected by the LiveCodeBench custom evaluator.
+            See: https://github.com/LiveCodeBench/LiveCodeBench?tab=readme-ov-file#custom-evaluation
+    """
+
+    model: str
+    lcb_input_files: list[str]
+
+
+def main() -> None:
+    model_to_lcb_input_files: dict[str, list[str]] = {
+        model: glob.glob(f"data/{model}/post-processed/*lcb-formatted.json") for model in MODELS
+    }
+    model_evaluation_infos: list[LcbModelEvaluationInfo] = [
```

Comment:
My first reaction was that this list is not necessary; the map comprehension below can run
```python
for model, lcb_inputs in model_to_lcb_input_files.items()
```

However, I have a different idea.  Rather than creating a small number of `LcbModelEvaluationInfo` objects, each of which has a list of `lcb_input_files`, I suggest creating many `LcbModelEvaluationInfo` objects, each of whose `lcb_input_files` is a singleton.  Now, you only have to have one concurrency loop, and you can control total parallelism in one place rather than having to multiply two numbers to understand the overall parallelism.